### DISCUSSION TITLE
How to make autocomplete leave function signatures?

### DISCUSSION THREAD

**Discussion by elfenlaid** [helix-editor/helix#11287] [Q&A] [answered]
Hi everyone,

There's one thing I can't get around with autocompletion. It doesn't fill in function signature arguments and such by default.

I wonder if there's a setting somewhere in Helix that I missed or it's purely an LSP thing?

The first screenshot shows a completion popover with the function's signature. 

![Screenshot 2024-07-23 at 21 07 12](https://github.com/user-attachments/assets/a07de04b-4fec-4485-9bc5-d2a52a1f42e2)

The second screenshot is what happens if I choose the completion. All arguments became blank.

![Screenshot 2024-07-23 at 21 07 40](https://github.com/user-attachments/assets/ea7b3e66-bfee-4775-98f5-58fd062b0716)

**Comment by the-mikedavis** [comment, marked as answer]
It isn't implemented in master yet (see https://github.com/helix-editor/helix/pull/9801) but this needs tabstop support. With proper tabstop support you can overwrite or leave the labels the language server provides and jump between them

**Reply by elfenlaid** [reply]
Oohhh, got it. Many thanks for the reply, going to keep my eye on then #9801 🙇
